app/src/main/kotlin/com/wire/android/ui/common/Extensions.kt

```diff
@@ -68,10 +72,28 @@ fun Modifier.shimmerPlaceholder(
 @OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun Modifier.clickable(clickable: Clickable?) = clickable?.let {
+    val syncStateObserver = LocalSyncStateObserver.current
+    val context = LocalContext.current
     this.combinedClickable(
         enabled = clickable.enabled,
-        onClick = clickable.onClick,
-        onLongClick = clickable.onLongClick
+        onClick = remember(clickable.blockUntilSynced) {
+            {
+                if (clickable.blockUntilSynced && !syncStateObserver.isSynced)
+                    Toast.makeText(context, context.getString(R.string.label_wait_until_synchronised), Toast.LENGTH_SHORT).show()
+                else
+                    clickable.onClick()
+            }
+        },
+        onLongClick = clickable.onLongClick?.let { onLongClick ->
+            remember(clickable.blockUntilSynced) {
+                {
+                    if (clickable.blockUntilSynced && !syncStateObserver.isSynced)
+                        Toast.makeText(context, context.getString(R.string.label_wait_until_synchronised), Toast.LENGTH_SHORT).show()
+                    else
+                        onLongClick()
+                }
+            }
+        }
```

Comment:
should we extract the clickable remember to a val for coherence with other composables?